spec/requests/api/v1.0/orders_spec.rb

```diff
@@ -64,6 +64,23 @@
         expect(response).to have_http_status(:bad_request)
       end
     end
+
+    context "when the survey has changed" do
+      let(:archived) { false }
+
+      let!(:order_item) { create(:order_item, :order => order) }
+      let!(:service_plan) { create(:service_plan, :portfolio_item => order.order_items.first.portfolio_item) }
+
+      before { allow(Catalog::SurveyCompare).to receive(:changed?).with(service_plan).and_return(true) }
```

Comment:
This is kinda wonky, like, it works because `changed?` does get called, but that's more the inner workings of `Catalog::SurveyCompare`. This spec really shouldn't have any knowledge of that, it should probably be stubbing `any_changed?`, I think.

Actually, it probably shouldn't even do that, it should just be making sure that the `CreateRequestForAppliedInventories` is being called, because we're already testing what that actually does in the spec for it.